planetiler-core/src/main/java/com/onthegomap/planetiler/reader/osm/PolyFileReader.java

```diff
@@ -0,0 +1,82 @@
+package com.onthegomap.planetiler.reader.osm;
+
+import static com.onthegomap.planetiler.geo.GeoUtils.JTS_FACTORY;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.locationtech.jts.geom.Coordinate;
+import org.locationtech.jts.geom.CoordinateXY;
+import org.locationtech.jts.geom.MultiPolygon;
+import org.locationtech.jts.geom.Polygon;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PolyFileReader {
+  private static final Logger LOGGER = LoggerFactory.getLogger(PolyFileReader.class);
+
+  private PolyFileReader() {
+    throw new IllegalStateException("Utility class");
+  }
+
+  public static MultiPolygon parsePolyFile(String filePath) throws NumberFormatException, IOException {
+    if (filePath == null) {
+      return null;
+    }
+
+    File file = new File(filePath);
+    try (
+      BufferedReader br =
+        new BufferedReader(new FileReader(file))
+    ) {
+      boolean inRing = false;
+      String line;
+      List<Polygon> polygons = new ArrayList<>();
+      List<Coordinate> currentRing = null;
```

Comment:
FYI I made a `MutableCoordinateSequence` if you wanted to use that, just call `add(x, y)` then you can pass it into createLinearRing directly.